core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/variables/transformation/TransformationCompiler.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.accesscontrol.blocks.variables.transformation
+
+import eu.timepit.refined.auto._
+import tech.beshu.ror.accesscontrol.blocks.variables.transformation.TransformationCompiler.CompilationError
+import tech.beshu.ror.accesscontrol.blocks.variables.transformation.TransformationCompiler.CompilationError.{UnableToCompileTransformation, UnableToParseTransformation}
+import tech.beshu.ror.accesscontrol.blocks.variables.transformation.domain.{Function, FunctionAlias, FunctionDefinition}
+import tech.beshu.ror.accesscontrol.blocks.variables.transformation.parser.Parser
+
+class TransformationCompiler(compiler: ExpressionCompiler, useAliases: Boolean) {
+
+  def compile(transformationString: String): Either[CompilationError, domain.Function] = {
+    for {
+      expression <- Parser.parse(transformationString)
+        .left.map(error => UnableToParseTransformation(error.message))
+      function <- compiler.compile(expression, useAlias = useAliases)
+        .left.map(error => UnableToCompileTransformation(error.message))
+    } yield function
+  }
+}
+
+object TransformationCompiler {
+  sealed trait CompilationError
+  object CompilationError {
+    final case class UnableToParseTransformation(message: String) extends CompilationError
+    final case class UnableToCompileTransformation(message: String) extends CompilationError
+  }
+
+  def withoutAliases: TransformationCompiler = {
```

Comment:
ok, now I understand what the `TransformationCompiler` is and it looks like I was wrong when I told you to move it to the `EnvironmentConfig` class. 

In some contexts, we use the compiler with aliases and in some without aliases. The crucial thing is the list of supported functions. We want to have the same set of functions in all contexts. So, maybe we should have the list of supported functions in the `EnvironmentConfig` class and use it to create a proper `TransformationCompiler` in places when it's needed. WDYT?